app/views/directives/key-value-editor.html

```diff
@@ -0,0 +1,238 @@
+          ng-attr-key-value-editor-focus="{{grabFocus && $last}}">
+
+        <!-- name/key help block -->
+        <span
+          class="help-block key-validation-error"
+          ng-show="(forms.keyValueEditor[uniqueForKey(unique, $index)].$error.pattern)">
+          <span class="validation-text">{{ entry.keyValidatorError || keyValidatorError }}</span>
+          <span ng-if="entry.keyValidatorErrorTooltip || keyValidatorErrorTooltip" class="help action-inline">
+            <a
+              aria-hidden="true"
+              data-toggle="tooltip"
+              data-placement="top"
+              data-original-title="{{entry.keyValidatorErrorTooltip || keyValidatorErrorTooltip}}"
+              title="{{entry.keyValidatorErrorTooltip || keyValidatorErrorTooltip}}">
+              <i class="{{entry.keyValidatorErrorTooltipIcon || keyValidatorErrorTooltipIcon}}"></i>
+            </a>
+          </span>
+        </span>
+        <span
+          class="help-block key-min-length"
+          ng-show="(forms.keyValueEditor[uniqueForKey(unique, $index)].$error.minlength)">
+          <span class="validation-text">Minimum character count is {{keyMinlength}}</span>
+        </span>
+        <span
+          class="help-block key-validation-error"
+          ng-show="(forms.keyValueEditor[uniqueForKey(unique, $index)].$error.required)">
+          <span class="validation-text">{{keyRequiredError}}</span>
+        </span>
+      </div>
+
+      <!-- the value block -->
+      <div
+        class="form-group key-value-editor-input"
+        ng-class="forms.keyValueEditor[uniqueForValue(unique, $index)].$invalid ? 'has-error' : ''">
+
+        <label for="uniqueForValue(unique, $index)" class="sr-only">{{valuePlaceholder}}</label>
+
+        <div ng-if="(!entry.valueFrom)">
+          <input
+            type="text"
+            class="form-control"
+            ng-class="{ '{{setFocusValClass}}' : $last  }"
+            id="{{uniqueForValue(unique, $index)}}"
+            name="{{uniqueForValue(unique, $index)}}"
+            ng-attr-placeholder="{{ (!isReadonlyAny) && valuePlaceholder || ''}}"
+            ng-minlength="{{valueMinlength}}"
+            maxlength="{{valueMaxlength}}"
+            ng-model="entry.value"
+            ng-readonly="isReadonlyAny || isReadonlySome(entry.name) || entry.isReadonly"
+            ng-pattern="validation.val"
+            ng-required="!allowEmptyKeys && entry.value">
+        </div>
+
+        <div ng-if="entry.valueFrom">
+          <div ng-if="isReadonlyAny || entry.isReadonlyValue" class="faux-input-group">
+            <span
+              class="faux-form-control-addon {{entry.valueIcon}}"
+              aria-hidden="true"
+              data-toggle="tooltip"
+              data-placement="top"
+              data-original-title="{{entry.valueIconTooltip || valueIconTooltip}}"
+              title="{{entry.valueIconTooltip || valueIconTooltip}}"></span>
```

Comment:
while we are fixing things... can you remove the extra title attribute here, it is causing https://bugzilla.redhat.com/show_bug.cgi?id=1395079